presto-main/src/main/java/com/facebook/presto/operator/scalar/RowIndeterminateOperator.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Override
+    public ScalarFunctionImplementation specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        checkArgument(arity == 1, "Expected arity to be 1");
+        Type type = boundVariables.getTypeVariable("T");
+        Class<?> indeterminateOperatorClass = generateIndeterminate(type, functionRegistry);
+        MethodHandle methodHandle = methodHandle(indeterminateOperatorClass, "indeterminate", type.getJavaType());
+        return new ScalarFunctionImplementation(false, ImmutableList.of(false), methodHandle, isDeterministic());
+    }
+
+    private static Class<?> generateIndeterminate(Type type, FunctionRegistry functionRegistry)
+    {
+        CallSiteBinder binder = new CallSiteBinder();
+
+        // Embed the MD5 hash code of input and output types into the generated class name instead of the raw type names,
+        // which could prevent the class name from hitting the length limitation and invalid characters.
+        byte[] md5Suffix = Hashing.md5().hashBytes(type.toString().getBytes()).asBytes();
+
+        ClassDefinition definition = new ClassDefinition(
+                a(PUBLIC, FINAL),
+                CompilerUtils.makeClassName(Joiner.on("$").join("RowIndeterminateOperator", BaseEncoding.base16().encode(md5Suffix))),
+                type(Object.class));
+
+        Parameter value = arg("value", type.getJavaType());
+
+        MethodDefinition method = definition.declareMethod(
+                a(PUBLIC, STATIC),
+                "indeterminate",
+                type(boolean.class),
+                value);
+
+        Scope scope = method.getScope();
+        BytecodeBlock body = method.getBody();
+
+        Variable wasNull = scope.declareVariable(boolean.class, "wasNull");
+        body.append(wasNull.set(constantFalse()));
+
+        CachedInstanceBinder cachedInstanceBinder = new CachedInstanceBinder(definition, binder);
+
+        LabelNode end = new LabelNode("end");
+        List<Type> fieldTypes = type.getTypeParameters();
+        boolean unknownElement = fieldTypes.stream().anyMatch(t -> t.equals(UNKNOWN));
+        if (unknownElement) {
+            // if the current field type is UNKNOWN which means this field is null, directly return true
+            body.push(true)
+                    .append(wasNull.set(constantFalse()))
+                    .gotoLabel(end);
+        }
+        else {
+            for (int i = 0; i < fieldTypes.size(); i++) {
+                IfStatement ifNullField = new IfStatement("if the field is null...");
+                ifNullField.condition(value.invoke("isNull", boolean.class, constantInt(i)))
+                        .ifTrue(new BytecodeBlock()
+                                .push(true)
+                                .append(wasNull.set(constantFalse()))
+                                .gotoLabel(end));
+
+                if (hasNestedValues(fieldTypes.get(i))) {
```

Comment:
Why? You should call the operator recursively on every field, regardless of type. This class should have no specific knowledge about what types are "complex".